packages/x-charts-pro/src/context/ZoomProvider/ZoomProvider.tsx

```diff
@@ -1,26 +1,69 @@
 import * as React from 'react';
-import { ZoomContext } from './ZoomContext';
+import { AxisConfig, ScaleName, ChartsXAxisProps, ChartsYAxisProps } from '@mui/x-charts';
+import { cartesianProviderUtils } from '@mui/x-charts/internals';
+import { ZoomContext, ZoomState } from './ZoomContext';
+import { defaultizeZoom } from './defaultizeZoom';
+import { ZoomData } from './Zoom.types';
+
+const { defaultizeAxis } = cartesianProviderUtils;
 
 type ZoomProviderProps = {
   children: React.ReactNode;
+  /**
+   * The configuration of the x-axes.
+   * If not provided, a default axis config is used.
+   * An array of [[AxisConfig]] objects.
+   */
+  xAxis?: Partial<Pick<AxisConfig<ScaleName, any, ChartsXAxisProps>, 'id' | 'zoom'>>[];
+  /**
+   * The configuration of the y-axes.
+   * If not provided, a default axis config is used.
+   * An array of [[AxisConfig]] objects.
+   */
+  yAxis?: Partial<Pick<AxisConfig<ScaleName, any, ChartsYAxisProps>, 'id' | 'zoom'>>[];
 };
 
-export function ZoomProvider({ children }: ZoomProviderProps) {
-  const [zoomRange, setZoomRange] = React.useState<[number, number]>([0, 100]);
+export function ZoomProvider({ children, xAxis: inXAxis, yAxis: inYAxis }: ZoomProviderProps) {
   const [isInteracting, setIsInteracting] = React.useState<boolean>(false);
 
+  const options = React.useMemo(
+    () =>
+      [
+        ...(defaultizeZoom(defaultizeAxis(inXAxis, 'x'), 'x') ?? []),
+        ...(defaultizeZoom(defaultizeAxis(inYAxis, 'y'), 'y') ?? []),
+      ].reduce(
+        (acc, v) => {
+          acc[v.axisId] = v;
+          return acc;
+        },
+        {} as ZoomState['options'],
+      ),
+    [inXAxis, inYAxis],
+  );
+
+  const [zoomData, setZoomData] = React.useState<ZoomData[]>(() =>
+    Object.values(options).map(({ axisId, start, end }) => ({ axisId, start, end })),
+  );
```

Comment:
Yeah I didn't really give it much attention because I wanted to do the controlling in different PR and since this is not released yet I didn't see much value in figuring out the best approach.

The thing with `initialStart/End` is that they are both initial and a limit. Since they are the limit, and the values aren't currently controlled, then they are the actual min/max the chart can reach.